### PR TITLE
Adds the first KerasCV tutorial

### PR DESCRIPTION
**Pull request by LukeWood**
Let's review this here, then we can merge it after all of the autogenerated docs go through

### CODE REVIEW

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+"""
+## Data Loading
+
+This guide uses the oxford_flowers102 dataset for demonstration purposes.
+
+To get started, we will first load the dataset:
+"""
+
+AUTOTUNE = tf.data.AUTOTUNE
+data, ds_info = tfds.load("oxford_flowers102", with_info=True, as_supervised=True)
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:52]: You probably want to disable the tfds progress bar first.

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+
+def prepare(image, label):
+    image = tf.image.resize(image, IMAGE_SIZE)
+    image = tf.cast(image, tf.float32)
+    label = tf.one_hot(label, num_classes)
+    return {"images": image, "labels": label}
+
+
+def prepare_dataset(ds, split):
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:73]: Avoid abbreviations, e.g. use `dataset` instead of `ds`

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+We will also begin by import all required packages:
+"""
+
+import keras_cv
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from keras_cv.layers import preprocessing
+from tensorflow.keras import applications, losses, optimizers
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:41]: One import per line

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+```
+
+We will also begin by import all required packages:
+"""
+
+import keras_cv
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from keras_cv.layers import preprocessing
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:40]: We should avoid nested namespaces. All layers should be imported from `keras_cv.layers`. To avoid confusion with `keras.layers`, it will be helpful to keep the `keras_cv.layers` prefix for every usage in this file (so just `import keras_cv` here)

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+
+To remedy this issue KerasCV offers RandAugment, a general purpose image data
+augmentation algorithm.  RandAugment has been shown to provide improved image
+classification results across numerous datasets, and only has two hyperparameters to
+configure.
+
+Let's check out KerasCV's RandAugment implementation.
+"""
+
+rand_augment = preprocessing.RandAugment(
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:132]: `keras_cv.layers.RandAugment`

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+"""
+Finally we train the model:
+"""
+
+strategy = tf.distribute.MirroredStrategy()
+with strategy.scope():
+    model = get_model()
+    model.fit(
+        train_ds,
+        steps_per_epoch=1000,
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:365]: Do we really need this argument?

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+KerasCV!
+
+As an additional exercise for readers, you can:
+
+- Perform a hyper parameter search over the RandAugment parameters to improve the
+classifier accuracy
+- Substitute the Oxford Flowers dataset with your own dataset
+- Experiment with custom `RandomAugmentationPipeline`s.
+
+Currently, between Keras core and KerasCV there are [_28 image augmentation
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:383]: All mdown links should be formatted on a single line (cf [guide](https://github.com/keras-team/keras-io/blob/master/contributor_guide.md))

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+- (Optional) `probability_to_apply` indicates the probability to apply the augmentation
+applied at each layer.
+
+That's all!  Let's apply our RandAugment layer:
+"""
+
+
+@tf.function
+def apply_rand_augment(inputs):
+    result = inputs.copy()
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:159]: Is this line really necessary?

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+- (Optional) `magnitude_standard_deviation` allows `magnitude` to be randomly sampled
+from a distribution with a standard deviation of `magnitude_standard_deviation`
+- (Optional) `probability_to_apply` indicates the probability to apply the augmentation
+applied at each layer.
+
+That's all!  Let's apply our RandAugment layer:
+"""
+
+
+@tf.function
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:157]: Shouldn't `tf.function` happen inside `RandomAugment` directly, so that users don't have to set it up by hand?

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+To remedy this issue KerasCV offers RandAugment, a general purpose image data
+augmentation algorithm.  RandAugment has been shown to provide improved image
+classification results across numerous datasets, and only has two hyperparameters to
+configure.
+
+Let's check out KerasCV's RandAugment implementation.
+"""
+
+rand_augment = preprocessing.RandAugment(
+    value_range=(0, 255),
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:133]: This is worth a comment / explanation

File: examples/keras_cv/data_augmentation_with_keras_cv.py [hunk 11]
```diff
@@ -0,0 +1,388 @@
+"""
+Title: State of the Arts Image Data Augmentation with KerasCV
```
Comment (fchollet) [inline examples/keras_cv/data_augmentation_with_keras_cv.py:2]: "CutMix, MixUp, and RandAugment image augmentation with KerasCV"

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Do you have a live Colab notebook version so we can check out the visualizations?

File: examples/keras_cv/data_augmentation_with_keras_cv.py (hunk 8 above)
Comment (LukeWood) [reply examples/keras_cv/data_augmentation_with_keras_cv.py:159]: Unfortunately it is and I'm not really sure why.  It's really strange to me; but I get an error saying that tf functions may not modify their inputs.  The weirder thing, this doesn't happen when I tf.function-ify in my classes' call methods.

File: examples/keras_cv/data_augmentation_with_keras_cv.py (hunk 9 above)
Comment (LukeWood) [reply examples/keras_cv/data_augmentation_with_keras_cv.py:157]: Good point.  I'll do this.

File: examples/keras_cv/data_augmentation_with_keras_cv.py (hunk 8 above)
Comment (fchollet) [reply examples/keras_cv/data_augmentation_with_keras_cv.py:159]: Definitely something we should clear up, since it's likely to be an error that users will commonly run into, and they may not figure out the workaround on their own.

**Comment by LukeWood** [comment]
https://colab.research.google.com/drive/1t3lwXhA1guK3nwIyz1-kIs84OuZUMci5?usp=sharing 

@fchollet

**Comment by LukeWood** [comment]
Updated in https://github.com/keras-team/keras-io/pull/852

Closing in favor of that.  Addressed all of your comments in that one.
